### PR TITLE
Feat/error handling

### PR DESCRIPTION
**Pull request by Denzel-Turambi**
What I did:
Did this break anything?
- [x] No

Type of change
- [x] New feature (non-breaking change which adds functionality)
- Added error handling for 500 and 404 errors
- Implemented PropTypes to ensure data types

Checklist:
- [x]  The code runs locally
- [x]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing
What's next?
Accessibility and Cypress Testing

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,16 +4,25 @@ import { getFishData } from './ApiCalls';
...
 
 function App() {
   const [fishes, setFishes] = useState([]);
   const [saved, setSaved] = useState([]);
+  const [error, setError] = useState("");
 
   useEffect(() => {
     getFishData()
     .then(data => setFishes(data))
+    .catch(error => {
```
Comment (jmartin777) [inline src/App.js:19]: clean code . great to see how your passing the setError(error) state through your parent compnent.

File: src/App.js [hunk 2]
```diff
@@ -33,12 +46,14 @@ function App() {
           <Route path ="/" element={ 
             <Fishes fishes={fishes}/>
           } />
```
Comment (jmartin777) [inline src/App.js:48]: error and setError props are passed down to the <Focus> component correctly, which can use them for error handling when fetching data from the API or dealing with other errors. The Routes seem to implemented correctly

If im correct it seems the the last <Route path="*"> handles unknown routes and redirects them to the 404 page using the <Navigate> component.?

File: src/Focus.js [hunk 3]
```diff
@@ -5,15 +5,25 @@ import './Focus.css'
 
 function Focus(props) {
   const [selectedFish, setSelectedFish] = useState({})
-  const [savedBool, setSavedBool] = useState(false);
 
   const {name} = useParams();
 
   useEffect(() => {
     getSingleFish(name)
     .then(data => setSelectedFish(data))
```
Comment (jmartin777) [inline src/Focus.js:14]: Now passing the setError(error) state through your Focus.js component. Looks good!

**Review by jmartin777 — COMMENTED** [review]
1.  Card.js, PropTypes are added to validate the name, img, and id props.
2. Fishes.js, PropTypes are added to validate the fishes prop.
3. Focus.js, PropTypes are added to validate the toggleSaved, saved, error, and setError props.
4.SavedFishCard.js, PropTypes are added to validate the name, img, id, and deleteSaved props.
5.SavedFishes.js, PropTypes are added to validate the saved and setSaved props.
 
 -Clear and concise documentation implementing Prop-Types throughout the project!

File: src/App.js (hunk 2 above)
Comment (Denzel-Turambi) [reply src/App.js:48]: Yea! The last routes redirects the users to a 404 error page whenever the user somehow reaches an unknown route.

**Pull request merged**